front/lib/swr.ts

```diff
@@ -421,16 +421,19 @@ export function useDataSourceContentNodes({
   internalIds,
 }: {
   owner: WorkspaceType;
-  dataSource: DataSourceType;
+  dataSource?: DataSourceType;
   internalIds: string[];
 }): {
   nodes: GetContentNodeResponseBody["nodes"];
   isNodesLoading: boolean;
   isNodesError: boolean;
 } {
-  const url = `/api/w/${owner.sId}/data_sources/${encodeURIComponent(
-    dataSource.name
-  )}/managed/content_nodes`;
+  const url =
+    dataSource && internalIds.length > 0
+      ? `/api/w/${owner.sId}/data_sources/${encodeURIComponent(
```

Comment:
Are we sure that this endpoint does not return anything if there is no `internalIds` provided?

Reply:
it returns `{}` , will check this does not cause any issue indeed :+1: